### DISCUSSION TITLE
Console help

### DISCUSSION THREAD

**Discussion by zejulio** [arendst/Tasmota#16440] [Q&A] [answered]
Hello,

I would like to have a console in my own DIY projects of ESP32 like tasmota has.
Can anyone point me in the right direction?
First with an introduction how it's done, and where can I check it on tasmota code.

Thanks!

**Comment by s-hadinger** [comment]
Sorry for the obvious answer, but why don't you just use Tasmota?

**Reply by zejulio** [reply]
Of course that I use Tasmota for what Tasmota has been designed. Home control and other stuff. I use it and like it.
But there are millions of things you can do with ESP devices that Tasmota is too much or I don't know how to so it.
Other answer (maybe the most obvious but none really wants to say) I like to do it myself 😉. And learn with others, in this case Tasmota guys that know way more than I. It's a good place to learn.
I'm kind of lost in the code, and I would like a guide to start.

**Comment by s-hadinger** [comment, marked as answer]
Look for `HandleConsole` function in `xdrv_01_9_webserver.ino`. This is where the console is handled. But it uses a lot of other features of Tasmota.
